pubsublite/producer.go

```diff
@@ -161,17 +161,23 @@ func (p *Producer) ProcessBatch(ctx context.Context, batch *model.Batch) error {
 		if !ok {
 			return fmt.Errorf("pubsublite: unable to find producer for %s", topic)
 		}
-		responses = append(responses, producer.Publish(ctx, &msg))
-	}
-	// NOTE(marclop) should the error be returned to the client? Does it care?
-	for _, res := range responses {
-		if serverID, err := res.Get(ctx); err != nil {
-			p.cfg.Logger.Error("failed producing message",
-				zap.Error(err),
-				zap.String("server_id", serverID),
-			)
-		}
+		result := producer.Publish(ctx, &msg)
+		wg.Add(1)
+		go func() {
```

Comment:
This PR is about updating logging and error messages for consistency, this seems a much greater change than that. Could you revert these changes, please?

Reply:
Ah, this was to avoid storing tmp variables to retrieve topics and headers later on. I've changed it to a tmp variable